diff --git a/CMakeLists.txt b/CMakeLists.txt
index c5cd16d55..0480120ad 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -707,9 +707,11 @@ set_target_properties(subzero PROPERTIES FOLDER "third_party")
 ###########################################################
 # marl
 ###########################################################
-set(MARL_THIRD_PARTY_DIR ${THIRD_PARTY_DIR})
-add_subdirectory(${THIRD_PARTY_DIR}/marl)
-set_target_properties(marl PROPERTIES FOLDER "third_party")
+find_package(marl CONFIG REQUIRED)
+add_library(marl ALIAS marl::marl)
+#set(MARL_THIRD_PARTY_DIR ${THIRD_PARTY_DIR})
+#add_subdirectory(${THIRD_PARTY_DIR}/marl)
+#set_target_properties(marl PROPERTIES FOLDER "third_party")
 
 if(MARL_THREAD_SAFETY_ANALYSIS_SUPPORTED)
     list(APPEND SWIFTSHADER_COMPILE_OPTIONS "-Wthread-safety")
@@ -798,6 +800,9 @@ else()
     message(FATAL_ERROR "REACTOR_BACKEND must be 'LLVM', 'LLVM-Submodule' or 'Subzero'")
 endif()
 
+find_package(SPIRV-Tools CONFIG REQUIRED)
+find_package(SPIRV-Tools-opt CONFIG REQUIRED)
+add_library(SPIRV-Tools ALIAS SPIRV-Tools-static)
 if (NOT TARGET SPIRV-Tools)
     # This variable is also used by SPIRV-Tools to locate SPIRV-Headers
     set(SPIRV-Headers_SOURCE_DIR "${THIRD_PARTY_DIR}/SPIRV-Headers")
diff --git a/src/Pipeline/CMakeLists.txt b/src/Pipeline/CMakeLists.txt
index 374d42467..999d675d0 100644
--- a/src/Pipeline/CMakeLists.txt
+++ b/src/Pipeline/CMakeLists.txt
@@ -71,9 +71,9 @@ if (NOT TARGET SPIRV-Tools)
     message(FATAL_ERROR "Missing required target: SPIRV-Tools")
 endif()
 
-set_target_properties(core_tables PROPERTIES FOLDER "SPIRV-Tools build")
-set_target_properties(enum_string_mapping PROPERTIES FOLDER "SPIRV-Tools build")
-set_target_properties(extinst_tables PROPERTIES FOLDER "SPIRV-Tools build")
+#set_target_properties(core_tables PROPERTIES FOLDER "SPIRV-Tools build")
+#set_target_properties(enum_string_mapping PROPERTIES FOLDER "SPIRV-Tools build")
+#set_target_properties(extinst_tables PROPERTIES FOLDER "SPIRV-Tools build")
 
 set_target_properties(vk_pipeline PROPERTIES
     POSITION_INDEPENDENT_CODE 1
diff --git a/src/Vulkan/CMakeLists.txt b/src/Vulkan/CMakeLists.txt
index bd05b0fc8..708b0bb86 100644
--- a/src/Vulkan/CMakeLists.txt
+++ b/src/Vulkan/CMakeLists.txt
@@ -258,3 +258,6 @@ configure_file(
     "vk_swiftshader_icd.json.tmpl"
     "${CMAKE_BINARY_DIR}/${CMAKE_SYSTEM_NAME}/vk_swiftshader_icd.json"
 )
+
+install(FILES "${CMAKE_BINARY_DIR}/${CMAKE_SYSTEM_NAME}/vk_swiftshader_icd.json" DESTINATION share/swiftshader)
+install(TARGETS vk_swiftshader)
